src/nvidiautil@ethanwharris/processor.js

```diff
@@ -28,76 +29,81 @@ var OPTIMUS = 2;
-  parse(output) {
+class Processor {
+    // Abstract: true,
+    constructor(name, baseCall, tailCall) {
+        this._name = name;
+        this._baseCall = baseCall;
+        this._tailCall = tailCall;
+        this._call = this._baseCall;
+
+        this._parseFunction = function () {};
+    }
+
+    parse() {
     // Do Nothing
-  }
-  process() {
-    let call = this._call + this._tailCall;
-    Subprocess.execCommunicate(call.split(' '))
+    }
+
+    process() {
+        let call = this._call + this._tailCall;
+        Subprocess.execCommunicate(call.split(' '))
       .then(output => this.parse(output)).catch(e => {
-        let title = 'Execution of ' + call + ' failed:';
-        Main.notifyError(title, e.message);
+          let title = `Execution of ${call} failed:`;
+          Main.notifyError(title, e.message);
       });
-  }
-  addProperty(parseFunction, callExtension) {
-    this._call += callExtension;
-    this._parseFunction = andThen(this._parseFunction, parseFunction);
-  }
-  getName() {
-    return this._name;
-  }
+    }
+
+    addProperty(parseFunction, callExtension) {
+        this._call += callExtension;
+        this._parseFunction = andThen(this._parseFunction, parseFunction);
+    }
+
+    getName() {
+        return this._name;
+    }
 }
 
-var NvidiaSettingsProcessor = class extends Processor {
-  constructor() {
-    super('nvidia-settings', 'nvidia-settings ', '-t');
-  }
-  parse(output) {
-    this._parseFunction(output.split('\n'));
-  }
+class NvidiaSettingsProcessor extends Processor {
+    constructor() {
+        super('nvidia-settings', 'nvidia-settings ', '-t');
+    }
+
+    parse(output) {
+        this._parseFunction(output.split('\n'));
+    }
 }
 
-var OptimusSettingsProcessor = class extends Processor {
-  constructor() {
-    super('optirun nvidia-smi', 'optirun nvidia-smi --query-gpu=', ' --format=csv,noheader,nounits');
-  }
-  parse(output) {
-    let items = output.split('\n').map(line => line.split(',')).flat();
-    this._parseFunction(items);
-  }
+class OptimusSettingsProcessor extends Processor {
+    constructor() {
+        super('optirun nvidia-smi', 'optirun nvidia-smi --query-gpu=', ' --format=csv,noheader,nounits');
+    }
+
+    parse(output) {
+        let items = output.split('\n').map(line => line.split(',')).flat();
+        this._parseFunction(items);
+    }
 }
 
-var NvidiaSmiProcessor = class extends Processor {
-  constructor() {
-    super('nvidia-smi', 'nvidia-smi --query-gpu=', ' --format=csv,noheader,nounits');
-  }
-  parse(output) {
-    let items = output.split('\n').map(line => line.split(',')).flat();
-    this._parseFunction(items);
-  }
+class NvidiaSmiProcessor extends Processor {
+    constructor() {
+        super('nvidia-smi', 'nvidia-smi --query-gpu=', ' --format=csv,noheader,nounits');
+    }
+
+    parse(output) {
+        let items = output.split('\n').map(line => line.split(',')).flat();
+        this._parseFunction(items);
+    }
 }
 
 var LIST = [
```

Comment:
we should rename this variable, the existing name isn't particularly descriptive.

Reply:
Any smart ideas? `PROCESSORS` perhaps?